commit 489c0ed20155bd388a1a81a8e575a519c7ecda72
Author: Luke Melia <luke@lukemelia.com>
Date:   Fri May 29 09:56:41 2009 -0400

    Avoid loading the ActiveRecord::SessionStore class on initialization if it is not in use.

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 9ffb00f..9fe240d 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -442,7 +442,7 @@ Run `rake gems:install` to install the missing gems.
     def initialize_database_middleware
       if configuration.frameworks.include?(:active_record)
         if configuration.frameworks.include?(:action_controller) &&
-            ActionController::Base.session_store == ActiveRecord::SessionStore
+            ActionController::Base.session_store.name == 'ActiveRecord::SessionStore'
           configuration.middleware.insert_before :"ActiveRecord::SessionStore", ActiveRecord::ConnectionAdapters::ConnectionManagement
           configuration.middleware.insert_before :"ActiveRecord::SessionStore", ActiveRecord::QueryCache
         else
